panther_manager/src/docking_manager_node.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_manager/docking_manager_node.hpp"
+
+#include <any>
+#include <chrono>
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "ament_index_cpp/get_package_share_directory.hpp"
+#include "behaviortree_ros2/ros_node_params.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "panther_utils/moving_average.hpp"
+
+#include <panther_manager/behavior_tree_manager.hpp>
+#include <panther_manager/behavior_tree_utils.hpp>
+
+namespace panther_manager
+{
+
+DockingManagerNode::DockingManagerNode(
+  const std::string & node_name, const rclcpp::NodeOptions & options)
+: Node(node_name, options)
+{
+  RCLCPP_INFO(this->get_logger(), "Constructing node.");
+
+  DeclareParameters();
+  const std::map<std::string, std::any> docking_initial_bb = {};
+  const int bt_server_port = this->get_parameter("bt_server_port").as_int();
+
+  docking_tree_manager_ = std::make_unique<BehaviorTreeManager>(
+    "Docking", docking_initial_bb, bt_server_port);
+
+  RCLCPP_INFO(this->get_logger(), "Node constructed successfully.");
+}
+
+void DockingManagerNode::Initialize()
+{
+  RCLCPP_INFO(this->get_logger(), "Initializing.");
+
+  RegisterBehaviorTree();
+  docking_tree_manager_->Initialize(factory_);
+
+  using namespace std::placeholders;
+
+  const float timer_freq = this->get_parameter("timer_frequency").as_double();
```

Comment:
We get double and then convert to float.
```suggestion
  const double timer_freq = this->get_parameter("timer_frequency").as_double();
```

or

```suggestion
  const auto timer_freq = this->get_parameter("timer_frequency").as_double();
```